apps/omg/lib/omg/crypto.ex

```diff
@@ -21,46 +21,55 @@ defmodule OMG.Crypto do
   `OMG.DevCrypto` in `test/support`
   """
   alias OMG.Signature
+  alias OMG.State.Transaction
+  alias OMG.TypedDataSign
 
   @type sig_t() :: <<_::520>>
   @type pub_key_t() :: <<_::512>>
   @type priv_key_t() :: <<_::256>> | <<>>
   @type address_t() :: <<_::160>>
   @type hash_t() :: <<_::256>>
+  @type chain_id_t() :: pos_integer() | nil
+  @type domain_separator_t() :: <<_::256>> | nil
 
   @doc """
   Produces a cryptographic digest of a message.
   """
   def hash(message), do: message |> ExthCrypto.Hash.hash(ExthCrypto.Hash.kec())
 
   @doc """
-  Verifies if private key corresponding to `address` was used to produce `signature` for
-  this `msg` binary.
+  Verifies if signature was created by private key corresponding to `address` and structured data
+  used to sign was derived from `domain_separator` and `raw_tx`
+  NOTE: Used only in tests
   """
-  @spec verify(binary, binary, address_t()) :: {:ok, boolean}
-  def verify(msg, signature, address) do
-    {:ok, recovered_address} = msg |> hash() |> recover_address(signature)
+  @spec verify(Transaction.t(), sig_t(), address_t(), chain_id_t(), domain_separator_t()) :: {:ok, boolean}
```

Comment:
if this is used only in tests, why isn't it defined in tests?

Reply:
Or maybe even stronger:
1. let's ditch the usage of this in `CryptoTest`, and just use the usual `hash |> recover |> compare` there
2. let's use `struct_hash |> recover |> compare` in the newly added tests as well

Then we can get rid of this function altogether

Reply:
and what about this too?